java/src/main/java/com/amido/stacks/workloads/menu/api/v1/MenuController.java

```diff
@@ -0,0 +1,206 @@
+    return new ResponseEntity<>(
+        new ResourceCreatedResponse(
+            createMenuHandler
+                .handle(requestToCommandMapper.map(correlationId, body))
+                .orElseThrow()),
+        HttpStatus.CREATED);
+  }
+
+  @GetMapping
+  @Operation(
+      tags = "Menu",
+      summary = "Get or search a list of menus",
+      description =
+          "By passing in the appropriate options, you can search for available menus in the system")
+  @ApiResponse(
+      responseCode = "200",
+      description = "Success",
+      content =
+          @Content(
+              mediaType = "application/json",
+              schema = @Schema(implementation = SearchMenuResult.class)))
+  @SearchAPIResponses
+  ResponseEntity<SearchMenuResult> searchMenu(
+      @RequestParam(value = "searchTerm", required = false) String searchTerm,
+      @RequestParam(value = "restaurantId", required = false) UUID restaurantId,
+      @RequestParam(value = "pageSize", required = false, defaultValue = "20") Integer pageSize,
+      @RequestParam(value = "pageNumber", required = false, defaultValue = "1")
+          Integer pageNumber) {
+
+    List<Menu> menuList;
+
+    if (isNotEmpty(searchTerm) && nonNull(restaurantId)) {
+      menuList =
+          menuQueryService.findAllByRestaurantIdAndNameContaining(
+              restaurantId, searchTerm, pageSize, pageNumber);
+    } else if (isNotEmpty(searchTerm)) {
+      menuList = menuQueryService.findAllByNameContaining(searchTerm, pageSize, pageNumber);
+    } else if (nonNull(restaurantId)) {
+      menuList = menuQueryService.findAllByRestaurantId(restaurantId, pageSize, pageNumber);
+    } else {
+      menuList = menuQueryService.findAll(pageNumber, pageSize);
+    }
+
+    return ResponseEntity.ok(
+        new SearchMenuResult(
+            pageSize,
+            pageNumber,
+            menuList.stream()
+                .map(m -> searchMenuResultItemMapper.toDto(m))
+                .collect(Collectors.toList())));
+  }
+
+  @GetMapping(value = "/{id}")
+  @Operation(
+      tags = "Menu",
+      summary = "Get a menu",
+      description =
+          "By passing the menu id, you can get access to available categories and items in the menu")
+  @ApiResponse(
+      responseCode = "200",
+      description = "Menu",
+      content =
+          @Content(
+              mediaType = MediaType.APPLICATION_JSON_VALUE,
+              schema = @Schema(implementation = MenuDTO.class)))
+  @ReadAPIResponses
+  ResponseEntity<MenuDTO> getMenu(
+      @PathVariable(name = "id") UUID id,
+      @Parameter(hidden = true) @RequestAttribute("CorrelationId") String correlationId) {
+
+    Menu menu =
+        menuQueryService
+            .findById(id)
+            .orElseThrow(() -> new MenuNotFoundException(new MenuCommand(correlationId, id)));
+    return ResponseEntity.ok(menuMapper.toDto(menu));
+  }
+
+  @PutMapping(
+      value = "/{id}",
+      consumes = MediaType.APPLICATION_JSON_VALUE,
```

Comment:
Do we actually need these produces and consumes for it to work?

Reply:
done - tested without produces & consumes and it worked fine.